homeassistant/components/notify/discord.py

```diff
@@ -42,13 +44,21 @@ def async_send_message(self, message, **kwargs):
         import discord
         discord_bot = discord.Client(loop=self.hass.loop)
 
+        # pylint: disable=unused-variable
         @discord_bot.event
         @asyncio.coroutine
-        def on_ready():  # pylint: disable=unused-variable
+        def on_ready():
             """Send the messages when the bot is ready."""
-            for channelid in kwargs[ATTR_TARGET]:
-                channel = discord.Object(id=channelid)
-                yield from discord_bot.send_message(channel, message)
+            try:
+                for channelid in kwargs[ATTR_TARGET]:
+                    channel = discord.Object(id=channelid)
+                    yield from discord_bot.send_message(channel, message)
+            except (discord.errors.HTTPException,
+                    discord.errors.NotFound) as error:
+                _LOGGER.warning("Communication error: %s", error)
+            # pylint: disable=broad-except
+            except KeyError:
```

Comment:
Which access can result it KeyError?

If  `kwargs[ATTR_TARGET]` then it can be checked in the beginning of async_send_message.